File: lib/templates/modal.ml
open Tyxml_html

let modal_ok ~a id =
  button
    ~a:
      ([
         a_id id;
         a_class [ "btn"; "modal-btn"; "ok-btn" ];
         a_text_value "default";
       ]
      @ a)
    [ b [ txt "Ok" ] ]

let modal_cancel id =
  button
    ~a:
      [
        a_id id;
        a_class [ "btn"; "modal-btn"; "cancel-btn" ];
        a_text_value "";
        Unsafe.string_attrib "formmethod" "dialog";
      ]
    [ b [ txt "Cancel" ] ]

let modal id els =
  dialog ~a:[ a_id id; a_class [ "modal" ]; a_role [ "dialog" ] ] els

File: src/templates/modal.rs
use maud::{html, Markup};

pub fn modal_ok(id: &str) -> Markup {
    html! {
        button #(id) .btn.modal-btn.ok-btn value="default" {
            b { "Ok"; }
        }
    }
}

pub fn modal_cancel(id: &str) -> Markup {
    html! {
        button #(id) .btn.modal-btn.cancel-btn value="" formmethod="dialog" {
            b { "Cancel"; }
        }
    }
}

pub fn modal(id: &str, content: Markup) -> Markup {
    html! {
        dialog #(id) .modal role="dialog" {
            (content);
        }
    }
}
